internal/query/projection/member.go

```diff
@@ -0,0 +1,98 @@
+package projection
+
+import (
+	"github.com/caos/zitadel/internal/eventstore/handler"
+	"github.com/caos/zitadel/internal/eventstore/handler/crdb"
+	"github.com/caos/zitadel/internal/repository/member"
+)
+
+const (
+	MemberUserIDCol = "user_id"
+	MemberRolesCol  = "roles"
+
+	MemberCreationDate  = "creation_date"
+	MemberChangeDate    = "change_date"
+	MemberSequence      = "sequence"
+	MemberResourceOwner = "resource_owner"
+)
+
+type reduceMemberConfig struct {
+	cols  []handler.Column
+	conds []handler.Condition
+}
+
+type reduceMemberOpt func(reduceMemberConfig) reduceMemberConfig
+
+func withCol(col string, value interface{}) reduceMemberOpt {
```

Comment:
maybe a better func name, to make clear, that this is only for members (because its in the general pkg)